### PR TITLE
Create rewrite-modules.js

### PR DESCRIPTION
**Pull request by AlexJuarez**
ported rewrite modules to babel 6

### CODE REVIEW

File: scripts/babel-6/rewrite-modules.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        );
+      }
+    }
+  }
+
+  /**
+   * Transforms either individual or chained calls to `jest.dontMock('Foo')`,
+   * `jest.mock('Foo')`, and `jest.genMockFromModule('Foo')`.
+   */
+  function transformJestCall(context, call) {
```
Comment (steveluscher) [inline scripts/babel-6/rewrite-modules.js]: Can you test this with the following code, and see what it produces?

```
jest.mock('foo').dontMock('foo').genMockFromModule('foo');
```

I get:

```
jest.mock('./././foo').dontMock('././foo').genMockFromModule('./foo');
```

Not that you broke it, but I'm pretty sure this never worked.

File: scripts/babel-6/rewrite-modules.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    ) {
+      return undefined;
+    }
+    var moduleArg = call.arguments[0];
+    if (moduleArg && moduleArg.type === 'Literal') {
+      var module = mapModule(context, moduleArg.value);
+      if (module) {
+        return t.callExpression(
+          call.callee,
+          [t.literal(module)]
```
Comment (steveluscher) [inline scripts/babel-6/rewrite-modules.js]: The Babel 6 version of this is `t.stringLiteral`.

File: scripts/babel-6/rewrite-modules.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      t.memberExpression(object, member.property),
+      args
+    );
+  }
+
+  return {
+    visitor: {
+      CallExpression: {
+        exit(path) {
+          return (
```
Comment (steveluscher) [inline scripts/babel-6/rewrite-modules.js]: Visitor methods in Babel 6 plugins don't need return values anymore; just operate on the nodes and exit.

**Comment by steveluscher** [comment]
Take a look at @fkling's awesome AST explorer. You can now switch it into Babel 6 mode; simply copy and paste your plugin in, and hack until it covers every case! http://felix-kling.de/esprima_ast_explorer/#/a0H3ntJzOW

**Comment by AlexJuarez** [comment]
I noticed I made quite a few mistakes when I wrote some of the transition. I have a better version forth coming.

**Comment by AlexJuarez** [comment]
Bump, whats needed to move forward?

**Comment by steveluscher** [comment]
I'd love to see a few test cases that cover all the cases. I was able to copy and paste the latest implementation into the AST explorer and break it: http://astexplorer.net/#/VGbbBZzB7V/2

**Comment by zpao** [comment]
Please include a test plan in your description above. While we don't have explicit tests for these, it would be good to know what exactly was done to ensure that the transform works.

**Comment by AlexJuarez** [comment]
Ok, so I have reworked the code a little bit to use path.traverse. Additionally I have added tests for the new plugin to verify that it works.

**Comment by steveluscher** [comment]
It works! Leaving to @zpao to merge.

**Comment by steveluscher** [comment]
yh31dhh9do-4

File: scripts/babel-6/rewrite-modules.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  return {
+    visitor: {
+      CallExpression: {
+        exit(path, state) {
+          if (path.node.seen) {
+            return;
+          }
+          transformRequireCall(path, state);
+          transformJestCall(path, state);
+          path.node.seen = true;
```
Comment (steveluscher) [inline scripts/babel-6/rewrite-modules.js]: Just for your interest… I found `shouldSkip` in the handbook; maybe it's designed to do the same thing? https://github.com/thejameskyle/babel-plugin-handbook#paths

Comment (AlexJuarez) [inline scripts/babel-6/rewrite-modules.js]: I wonder if shouldSkip would skip all further transformations for that node, when I only want to skip the one I have defined.

File: package.json [hunk 5]
```diff
@@ -60,6 +61,7 @@
   },
   "dependencies": {
     "core-js": "^1.0.0",
+    "babel-core": "^6.3.17",
```
Comment (zpao) [inline package.json]: I'm going to followup and move this - it's not a dep of fbjs and really shouldn't even be a devdep (should be a devdep of fbjs-scripts, but we aren't running tests there right now so it's in the grey area)

**Pull request merged**